src/compositions/commons/Modal.tsx

```diff
@@ -13,19 +14,35 @@ interface ModalProps {
 
 export const Modal: VFC<ModalProps> = (props: ModalProps) => {
   const [isBrowser, setIsBrowser] = useState(false)
+  const modal = useRef<HTMLDivElement>(null)
 
   useEffect(() => {
     setIsBrowser(true)
   }, [])
 
+  useEffect(() => {
+    if (!props.show) return
+
+    function closeModalOnClickAway(event: MouseEvent) {
```

Comment:
- extract this function somewhere like in **util** directory, because like this function will be used again someday.
- change to arrow function